src/main/java/bio/terra/service/common/gcs/GcsUriUtils.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.service.common.gcs;
+
+import bio.terra.service.dataset.exception.InvalidUriException;
+import org.apache.commons.lang3.StringUtils;
+
+public final class GcsUriUtils {
+  public static class GsUrlParts {
+    private String bucket;
+    private String path;
+    private boolean isWildcard = false;
+
+    public String getBucket() {
+      return bucket;
+    }
+
+    public GsUrlParts bucket(String bucket) {
+      this.bucket = bucket;
+      return this;
+    }
+
+    public String getPath() {
+      return path;
+    }
+
+    public GsUrlParts path(String path) {
+      this.path = path;
+      return this;
+    }
+
+    public boolean getIsWildcard() {
+      return isWildcard;
+    }
+
+    public GsUrlParts isWildcard(boolean isWildcard) {
+      this.isWildcard = isWildcard;
+      return this;
+    }
+  }
+
+  /**
+   * Parse a Google Cloud Storage URI into its component pieces
+   *
+   * @param uri of type gs://<bucket_name>/<file_path_inside_bucket>
+   * @return Object representing uri pieces
+   */
+  public static GsUrlParts parseBlobUri(String uri) {
```

Comment:
I was wondering why `BlobId` doesn't have this API already and it turns out that it does, as of a recent PR. https://github.com/googleapis/java-storage/issues/868 (in version 1.117.0)

Their version doesn't do all the validation ours does, though. We could use their code to create the `BlobId` and then do our validation afterwards.